packages/expo-doctor/src/checks/InstalledDependencyVersionCheck.ts

```diff
@@ -0,0 +1,38 @@
+import spawnAsync, { SpawnResult } from '@expo/spawn-async';
+
+import { DoctorCheck, DoctorCheckParams, DoctorCheckResult } from './checks.types';
+
+function isSpawnResult(result: any): result is SpawnResult {
+  return 'stderr' in result && 'stdout' in result && 'status' in result;
+}
+
+export class InstalledDependencyVersionCheck implements DoctorCheck {
+  description = 'Checking dependency versions for compatibility with the installed Expo SDK';
+
+  async runAsync({ projectRoot }: DoctorCheckParams): Promise<DoctorCheckResult> {
+    const issues: string[] = [];
+    const advice: string[] = [];
+
+    try {
+      // only way to check dependencies without automatically fixing them is to use interactive prompt
+      // In the future, we should add JSON output to npx expo install, check for support, and use that instead
+      await spawnAsync('sh', ['-c', 'echo "n" | npx expo install --check'], {
+        stdio: 'pipe',
+        cwd: projectRoot,
+      });
+    } catch (error: any) {
+      if (isSpawnResult(error)) {
+        issues.push(error.stderr.trim());
+        advice.push(`Use npx expo install --check to review and upgrade your dependencies.`);
+      } else {
+        throw error;
```

Comment:
What's the difference between `issues` and assertions from within the function?

Reply:
You mean why could it possibly throw even if normally issues are logged? The issues would be like "we successfully completed the check, and the check found issues with your project", and throwing would be like, "something unexpectedly broke while attempting the check, let's bail". So you don't have to think about every possible error while writing a check, just the issues that your check is looking for. 

Generally this would mean writing a check wouldn't need any try/ catches, but this one breaks that pattern a bit, since `spawnAsync` throws on exit code !== 0 (e.g., the issues we're looking for), but in theory could also throw for another reason. Could definitely at least clarify in a comment.